src/prefect/futures.py

```diff
@@ -263,11 +272,23 @@ async def _get_state(self, client: OrionClient = None) -> State[R]:
         self.task_run = task_run
         return task_run.state
 
+    async def _wait_for_submission(self):
+        import asyncio
+
+        # TODO: This spin lock is not performant but is necessary for cases where a
+        #       future is created in a separate event loop i.e. when a sync task is
+        #       called in an async flow
+        if not asyncio.get_running_loop() == self._loop:
+            while not self.task_run:
+                await anyio.sleep(0)
+        else:
+            await self._submitted.wait()
+
     def __hash__(self) -> int:
-        return hash(self.run_key)
+        return hash(self.key)
 
     def __repr__(self) -> str:
-        return f"PrefectFuture({self.run_key!r})"
+        return f"PrefectFuture({self.name!r})"
```

Comment:
👏 This should make for more readable output.

Reply:
It's the same content actually, just a clearer usage.

Reply:
Oh, it seemed like it was dropping some other parts, but better name is good too.